### PR TITLE
[pytest] Convert interface naming mode test to pytest

### PR DESCRIPTION
**Pull request by ArunSaravananBalachandran**
### Description of PR

- Pytest conversion of interface naming mode test cases available in ansible/roles/test/tasks/iface_naming_mode.yml
- Add support for 'become_user' argument in AnsibleHostBase to enable logging in as different user to run the interface naming mode tests.

**Note:** Requires [pytest-ansible/PR46](https://github.com/ansible/pytest-ansible/commit/d33c025f070a9c870220a157cc5a999fda68de44) changes to support 'become' and 'become_user'.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### How did you verify/test it?

Logs: [IFACE_pytest_t1_topo_logs.txt](https://github.com/Azure/sonic-mgmt/files/4680656/IFACE_pytest_t1_topo_logs.txt), [IFACE_pytest_t0_topo_logs.txt](https://github.com/Azure/sonic-mgmt/files/4680703/IFACE_pytest_t0_topo_logs.txt)

### CODE REVIEW

**Comment by lguohan** [comment]
can you modify the sonic-mgmt build to include that patch in the sonic-mgmt docker?

**Comment by ArunSaravananBalachandran** [comment]
> can you modify the sonic-mgmt build to include that patch in the sonic-mgmt docker?

[sonic-buildimage/PR4681](https://github.com/Azure/sonic-buildimage/pull/4681) has been raised to include the pytest-ansible patch in docker-sonic-mgmt.

File: tests/iface_namingmode/test_iface_namingmode.py [hunk 1]
```diff
@@ -0,0 +1,829 @@
...
+    def test_show_lldp_table(self, setup, setup_config_mode, lldp_interfaces):
+        """
+        Checks whether 'show lldp table' lists the interface name as per
+        the configured naming mode
+        """
+        dutHostGuest, mode, ifmode = setup_config_mode
+        minigraph_neighbors = setup['minigraph_facts']['minigraph_neighbors']
+
+        lldp_table = dutHostGuest.shell('SONIC_CLI_IFACE_MODE={} show lldp table'.format(ifmode))['stdout']
+        logger.debug('lldp_table:\n{}'.format(lldp_table))
```
Comment (wangxin) [inline tests/iface_namingmode/test_iface_namingmode.py:164]: This log message will only be output if logging level is DEBUG. When logging level is DEBUG, the previous call to the shell module would log the module execution result too. It looks like duplicated logs. Maybe you can consider raising the logging level to info here. Similar comments for other logger.debug calls later in the script.

File: tests/iface_namingmode/test_iface_namingmode.py [hunk 2]
```diff
@@ -0,0 +1,829 @@
...
+         'up_ports' : up_ports,
+         'upport_alias_list' : upport_alias_list
+    }
+
+    yield setup_info
+
+    logger.info('Reverting the port alias name in redis db to the actual values')
+    for item in default_interfaces:
+        port_alias_old = port_alias_facts['port_name_map'][item]
+        duthost.command('redis-cli -n 4 HSET "PORT|{}" alias {}'.format(item, port_alias_old))
```
Comment (wangxin) [inline tests/iface_namingmode/test_iface_namingmode.py:62]: The original ansible version of this testing reboot the DUT to recover after the testing is done. Is setting port alias to old names enough to recover or do we need to perform a reboot or config reload here?

**Comment by wangxin** [comment]
Since the corresponding ansible testing has been converted to pytest, could you please update the ansible/roles/test/tasks/iface_naming_mode.yml to call this pytest script? And rest of the unused ansible playbooks can be removed too.

File: tests/iface_namingmode/test_iface_namingmode.py (hunk 1 above)
Comment (ArunSaravananBalachandran) [reply tests/iface_namingmode/test_iface_namingmode.py]: The logging level has been changed from DEBUG to INFO.

File: tests/iface_namingmode/test_iface_namingmode.py (hunk 2 above)
Comment (ArunSaravananBalachandran) [reply tests/iface_namingmode/test_iface_namingmode.py:63]: All the configurations changes done as part of the testing are reverted by the script itself. So reboot or config reload is not required.

**Comment by ArunSaravananBalachandran** [comment]
> Since the corresponding ansible testing has been converted to pytest, could you please update the ansible/roles/test/tasks/iface_naming_mode.yml to call this pytest script? And rest of the unused ansible playbooks can be removed too.

@wangxin , the "ansible/roles/test/tasks/iface_naming_mode.yml" has been modified to call the pytest script and the other playbooks have been removed.

**Pull request merged**